engine/tests/ElggCoreGetEntitiesFromAnnotationsTest.php

```diff
@@ -330,13 +330,17 @@ public function testElggGetEntitiesFromAnnotationsCalculateConstrainedByWhere()
 			}
 		}
 
+		$prefix = _elgg_services()->db->getTablePrefix();
 		$options = array(
 			'type' => 'object',
 			'subtypes' => $subtypes,
 			'limit' => $num_entities,
 			'annotation_name' => $name,
 			'annotation_values' => array_unique(call_user_func_array('array_merge', $values)),
 			'calculation' => 'sum',
+			'joins' => array(
+				"JOIN {$prefix}metastrings msv ON (n_table.value_id = msv.id)",
```

Comment:
This user-given JOIN presents a problem in the COUNT query. Although the API does join the n_table in both cases, in the COUNT query the API adds the JOIN *after* this one, so the "n_table" reference isn't yet available.